src/stashcache.py

```diff
@@ -52,22 +42,31 @@ class NotRegistered(Exception):
 
 def _generate_ligo_dns() -> List[str]:
     """
-    Query the LIGO LDAP server for all grid DNs in the LVC collab.
+    Query the LIGO LDAP server for all grid DNs in the IGWN collab.
 
     Returns a list of DNs.
     """
-    ldap_obj = ldap.initialize("ldaps://ldap.ligo.org")
     query = "(&(isMemberOf=Communities:LSCVirgoLIGOGroupMembers)(gridX509subject=*))"
-    results = ldap_obj.search_s("ou=people,dc=ligo,dc=org", ldap.SCOPE_ONELEVEL,
-                                query, ["gridX509subject"])
-    all_dns = []
-    for result in results:
-        user_dns = result[1].get('gridX509subject', [])
-        for dn in user_dns:
-            if dn.startswith(b"/"):
-                all_dns.append(dn.replace(b"\n", b" ").decode("utf-8"))
-
-    return all_dns
+    try:
+        server = ldap3.Server(_ligo_ldap_url, connect_timeout=10)
+        conn = ldap3.Connection(server, receive_timeout=10)
+        conn.bind()
+    except ldap3.core.exceptions.LDAPExceptionError:
+        log.exception("Failed to connect to the LIGO LDAP")
+        return []
+
+    try:
+        conn.search(_ligo_ldap_branch,
+                    query,
+                    search_scope='LEVEL',
+                    attributes=['gridX509subject'])
```

Comment:
The only thing i'll mention here is that i remember _not_ getting back the stuff i wanted when i explicitly listed the attributes i wanted, which led me just to request `attributes=['*']` instead (which gave more than i cared about, but at least not less than i cared about).

I would suggest trying it out both ways and just confirm you are not missing anything you care about with the explicit `attributes` list.

Reply:
Here, i mean
https://github.com/opensciencegrid/topology/blob/master/src/webapp/cilogon_ldap.py#L23

Reply:
This worked for me --maybe the issue you ran into was with requesting multiple attrs?